### ISSUE TITLE
put_bucket_lifecycle_configuration fails with "Missing required header for this request: Content-MD5"

### ISSUE THREAD

**Issue by eagletmt** [awslabs/aws-sdk-rust#121]
### Version

```
├── aws-sdk-s3 v0.0.8-alpha (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   ├── aws-auth v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   │   ├── smithy-types v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   ├── aws-endpoint v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   ├── aws-types v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   └── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── aws-http v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   ├── aws-types v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── smithy-types v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── aws-hyper v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   ├── aws-auth v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── aws-endpoint v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── aws-http v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── aws-sig-auth v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   │   ├── aws-auth v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   │   ├── aws-types v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── protocol-test-helpers v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   ├── smithy-client v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   │   ├── protocol-test-helpers v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   │   ├── smithy-http-tower v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   │   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   │   ├── smithy-types v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── smithy-http-tower v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── smithy-types v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── aws-sig-auth v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── aws-types v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── smithy-types v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   └── smithy-xml v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
```

### Platform

`Linux reinforce 5.4.72-microsoft-standard-WSL2 #1 SMP Wed Oct 28 23:40:43 UTC 2020 x86_64 GNU/Linux`

### AWS Services

S3

### Description

put_bucket_lifecycle_configuration fails with "Missing required header for this request: Content-MD5".

I tried this code:

```rust
#[tokio::main]
async fn main() -> Result<(), aws_sdk_s3::Error> {
    tracing_subscriber::fmt::init();

    let client = aws_sdk_s3::Client::from_env();
    client
        .put_bucket_lifecycle_configuration()
        .bucket("eagletmt-test-bucket")
        .lifecycle_configuration(
            aws_sdk_s3::model::BucketLifecycleConfiguration::builder()
                .rules(
                    aws_sdk_s3::model::LifecycleRule::builder()
                        .id("Expire")
                        .status(aws_sdk_s3::model::ExpirationStatus::Enabled)
                        .filter(aws_sdk_s3::model::LifecycleRuleFilter::Prefix(
                            "".to_owned(),
                        ))
                        .expiration(
                            aws_sdk_s3::model::LifecycleExpiration::builder()
                                .days(1)
                                .build(),
                        )
                        .build(),
                )
                .build(),
        )
        .send()
        .await?;
    Ok(())
}
```

I expected to see this happen: No output

Instead, this happened: InvalidRequest error saying "Missing required header for this request: Content-MD5"

```
Jun 19 21:19:35.940 TRACE smithy_http_tower::dispatch: request=Request { method: PUT, uri: https://s3.ap-northeast-1.amazonaws.com/eagletmt-test-bucket?lifecycle, version: HTTP/1.1, headers: {"content-type": "application/xml", "content-length": "558", "host": "s3.ap-northeast-1.amazonaws.com", "user-agent": "aws-sdk-rust/0.1.0 os/linux lang/rust/1.53.0", "x-amz-user-agent": "aws-sdk-rust/0.1.0 api/s3/0.0.8-alpha os/linux lang/rust/1.53.0", "authorization": Sensitive, "x-amz-date": "20210619T121935Z", "x-amz-content-sha256": "d53525ac7372d7c8ce739729d5cc6c92b2404b89e81edbad139a2afc5f0610ee"}, body: SdkBody { inner: Once(Some(b"<LifecycleConfiguration xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\"><Rule xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\"><Expiration xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\"><Days xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\">1</Days></Expiration><ID xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\">Expire</ID><Filter xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\"><Prefix xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\"></Prefix></Filter><Status xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\">Enabled</Status></Rule></LifecycleConfiguration>")), retryable: true } }
Jun 19 21:19:36.109 TRACE smithy_http::middleware: response=Response { status: 400, version: HTTP/1.1, headers: {"x-amz-id-2": "E6ZkgKb4a0HgG8OG6cxAnclLUcyakqvawl8djyWTkF12EuXF0NgikUVPicsy6qf6L4B3p4/UDME=", "x-amz-id-2": "E6ZkgKb4a0HgG8OG6cxAnclLUcyakqvawl8djyWTkF12EuXF0NgikUVPicsy6qf6L4B3p4/UDME=", "x-amz-request-id": "ZHEYMG0XCYEC1057", "x-amz-request-id": "ZHEYMG0XCYEC1057", "date": "Sat, 19 Jun 2021 12:19:38 GMT", "content-type": "application/xml", "content-length": "246", "server": "AmazonS3", "connection": "close"}, body: SdkBody { inner: Streaming(Body(Streaming)), retryable: false } }
Error: Unhandled(Error { code: Some("InvalidRequest"), message: Some("Missing required header for this request: Content-MD5"), request_id: Some("ZHEYMG0XCYEC1057"), extras: {"s3_extended_request_id": "E6ZkgKb4a0HgG8OG6cxAnclLUcyakqvawl8djyWTkF12EuXF0NgikUVPicsy6qf6L4B3p4/UDME="} })
```

**Comment by rcoh** [comment]
Looks like we need to add support for https://awslabs.github.io/smithy/1.0/spec/core/behavior-traits.html#httpchecksumrequired-trait

**Comment by rcoh** [comment]
Fixed in smithy-rs, this will go out in tomorrows release. Thanks for the detailed bug report!

**Comment by jdisanti** [comment]
This is fixed in `v0.0.9-alpha`.
